feat: 更新 README，添加选项页面和自定义服务器功能

Lihewin opened the pull request:
#7 重新提交并解决冲突问题

Lihewin committed:
feat: 更新 README，添加选项页面和自定义服务器功能

qwertycxz commented on bg.js:
```diff
@@ -37,21 +44,58 @@ async function encrypt(domainWithPort) {
+            VpnPrefix = `${customServer.baseUrl}/${url.protocol.replace(':', '')}/webvpn`;
+            serverKey = customServer.encryptionKey || encryptionKey;
+        } else {
+            return; // 服务器配置错误
```
你觉得呢？感觉静默错误的确不是什么很好的主意。

qwertycxz commented on options.js:
```diff
@@ -0,0 +1,201 @@
+    if (encryptionKey.length !== 16) {
+        showStatus('❌ 加密密钥必须恰好16个字符', 'error');
+        return;
+    }
```
关于Magic Number，我倒是无所谓，看你想不想改。

qwertycxz commented on options.js:
```diff
@@ -0,0 +1,201 @@
+    // 恢复为默认值
+    await chrome.storage.local.set({
+        vpnServer: 'nuist',
+        customServers: {},
```
这里连带清空了所有的customServers，有必要吗？如果本意如此的话，那应该给出更严厉的警告。

qwertycxz commented on options.html:
```diff
@@ -0,0 +1,411 @@
+            </label>
+
+            <label class="option-label">
+                <input type="radio" name="vpnServer" value="njupt" checked>
```
看了看实际的代码，这里的`checked`似乎没有被用到？我们是根据CSS的`checked`选择器进行选择的，所以HTML的`checked`选择器没有用。直接删掉就好。

qwertycxz commented on options.js:
```diff
@@ -0,0 +1,201 @@
+        list.appendChild(item);
+
+        // 为新添加的单选框添加事件监听
+        const radio = item.querySelector('input[type="radio"]');
```
AI说得没错，这里的inline HTML导致删除功能完全失效。

整个功能完全失效，这里必须改。

但我不喜欢AI这里的修改，用JS去拼HTML很丑，建议使用`<template>`。

qwertycxz commented on options.html:
```diff
@@ -0,0 +1,411 @@
+                </div>
+
+                <div class="form-group">
+                    <label class="form-label">加密密钥</label>
```
无障碍设计……我无所谓，这里改不改看你。

要改的话，其他几个也一并改改。

qwertycxz commented on options.js:
```diff
@@ -0,0 +1,201 @@
+        radio.addEventListener('change', function () {
+            showStatus('已选择: ' + server.name, 'info');
+        });
+    });
```
没错。不过既然要把这一段挪到`<template>`，那这里就不用改了。

qwertycxz commented on options.js:
```diff
@@ -0,0 +1,201 @@
+    const customServers = settings.customServers || {};
+
+    // 生成唯一的服务器ID
+    const serverId = 'custom_' + Date.now();
```
更安全的加密选项？我也无所谓，改不改看你。

qwertycxz commented on options.js:
```diff
@@ -0,0 +1,201 @@
+
+// 保存按钮事件
+document.getElementById('saveBtn').addEventListener('click', async function () {
+    const selectedServer = document.querySelector('input[name="vpnServer"]:checked').value;
```
似乎没关系，因为我们保证这里一定会选中。

qwertycxz commented on options.js:
```diff
@@ -0,0 +1,201 @@
+}
+
+// 添加自定义服务器
+document.getElementById('addServerBtn').addEventListener('click', async function () {
```
严重的问题：这里没有强制要求密钥长度（全局密钥那里约束了但这里没有），导致静默错误，需要修改。

qwertycxz reviewed (changes requested):
LGTM，就是有不少BUG……

Vibe Coding一时爽，Debug火葬场。

qwertycxz commented:
粗略看了看，可以先改着……

Code Review得有点乱……建议去[Files changed](https://github.com/qwertycxz/NUIST-VPN/pull/8/files)那里看。

Lihewin commented:
感谢review，我只粗略测试了一下。过段时间不忙了来修改。

Lihewin committed:
feat: 增强安全性和功能完善

- 实现密钥Base64编码存储机制，提升安全性
- 替换innerHTML为DOM API构建，防止XSS漏洞
- 优化密钥验证逻辑（AES-128 16字节）
- 使用UUID替代时间戳生成服务器唯一标识
- 完善异常处理和错误通知系统
- 修复默认VPN服务器为NUIST
- 添加向后兼容性支持旧版密钥格式
- 更新manifest.json权限配置和安全策略
- 生成gitignore文件

Lihewin committed:
feat: 添加自定义服务器项模板并优化服务器渲染逻辑、添加空选项检查

Lihewin committed:
feat: 添加 URL 检查功能以限制扩展页面和内置页面的访问，更新通知图标

Lihewin commented:
我尝试解决了部分问题并手动测试了一些用例。不过由于我并不熟练前端，所以难以彻底审查代码。如果仍有安全性问题请协助修改，谢谢。

qwertycxz committed:
chore: 移除无用的`checked`

qwertycxz commented on options.js:
```diff
@@ -0,0 +1,284 @@
+            vpnServer: 'nuist',
+            customServers: {},
+            encryptionKey: defaultKey
+        });
```
胡说八道什么呢

qwertycxz commented on bg.js:
```diff
@@ -14,44 +36,183 @@ function ab2hex(buffer) {
+            VpnPrefix = `https://client.vpn.nuist.edu.cn/${url.protocol.replace(':', '')}/webvpn`;
+        } else {
+            // 默认 njupt
+            VpnPrefix = `https://vpn.njupt.edu.cn:8443/${url.protocol.replace(':', '')}/webvpn`;
```
必要性不大

qwertycxz commented on options.js:
```diff
@@ -0,0 +1,284 @@
+    if (!url.startsWith('http://') && !url.startsWith('https://')) {
+        showStatus('❌ 地址必须以 http:// 或 https:// 开头', 'error');
+        return;
+    }
```
必要性不大

qwertycxz commented:
很抱歉过了两个月才回来，近日实在较忙。

其实我比较疑惑一个点：这些密钥真的需要 Base64 存储吗？Base64 不是加密，直接明文存储在 Local Storage 里我看并无不妥？现在这 Vibe Coding 搞出了一大堆冗余代码，全都是围绕着 Base64 出来的。

你旧版的 NJUPT VPN 没有使用 Local Storage 存密钥吧？没有的话我就不用考虑兼容性，直接把所有 Base64 相关的东西都删掉就好。

如果你想改可以来改，不想的话也可以我来，当然得过个一段时间……

Lihewin commented:
> 很抱歉过了两个月才回来，近日实在较忙。
> 
> 其实我比较疑惑一个点：这些密钥真的需要 Base64 存储吗？Base64 不是加密，直接明文存储在 Local Storage 里我看并无不妥？现在这 Vibe Coding 搞出了一大堆冗余代码，全都是围绕着 Base64 出来的。
> 
> 你旧版的 NJUPT VPN 没有使用 Local Storage 存密钥吧？没有的话我就不用考虑兼容性，直接把所有 Base64 相关的东西都删掉就好。
> 
> 如果你想改可以来改，不想的话也可以我来，当然得过个一段时间……

我觉得不base64也可以，只是不希望这个密钥在github中被明文检索到……

qwertycxz commented:
我正在着手重构这个PR，应该过两天就写好了😀

> 我觉得不base64也可以，只是不希望这个密钥在github中被明文检索到……

这个我觉得可以有👍我准备把默认密钥以base64存储，用户自定义还是直接明文

qwertycxz committed:
refactor: 重构项目以使用TypeScript

1. 将JavaScript代码迁移到TypeScript；
2. 将Vibe Coding的冗余代码整理，并尽量使用现代语法特性；
3. 使用Prettier格式化代码；
4. 设置界面增加无障碍标签与夜间模式；
5. 使用sync存储替代local存储。

TODO:
- [] 使用Bun进行构建；
- [] ESLint代码检查；
- [] GitHub Actions集成。

qwertycxz commented:
吼吼，这下可真是轰轰烈烈的大改进了。

欢迎来测试和 Code Review 😀。

`bun run build`就可以编译成js运行了（不喜欢bun的话，`npm run build`也行）

接下来要忙一段时间的考试了，忙完再回来继续搞~

qwertycxz committed:
chore: 更新依赖
